lib/Default/SmrPlanet.class.inc

```diff
@@ -680,7 +678,7 @@ class SmrPlanet {
 		
 		//trim buildings first
 		$this->setMaxBuildings();
-		foreach ($this->buildings as $id => $amt) {
+		foreach ($this->getBuildings() as $id => $amt) {
```

Comment:
Strictly speaking this change isn't needed as you run the fetch function on the line before

Reply:
Oh, maybe this comment is referring to the `getMaxBuildings` call on the next line? If so, you're right, we could either use the bare attribute or omit `setMaxBuildings` (since the getter ensures that it is set).

Reply:
Yeah, I was referring to the `$this->setMaxBuildings();` which does the same work as swapping in `getBuildings` does